# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I'm choosing Lane 2: Refresh/Content Opportunity Scoring. This maps to a ranking/scoring task
type, "which pages should be reviewed first?" which fits how I want to work: producing a
ranked queue with reason codes a reviewer can inspect, not a black-box score. This lane also
lines up with what I want to prove in this internship: catching results that look good but are
quietly wrong, before they're acted on.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

What decision does this improve? Which pages a content reviewer should check first, out of far
more pages than they have time to manually review.

Who acts on the output, and what do they do? A content reviewer opens the top-ranked pages and
decides: refresh, expand, protect, prune, or monitor.

What does a wrong answer cost? A false positive wastes limited reviewer time on a page that
didn't need attention. A false negative lets a real declining or under-optimized page keep
losing visibility, unnoticed, for longer than necessary.

Why does data/ML help at all? With more pages than any team can manually check, and signals
that are individually messy but meaningful in combination, a plain single rule isn't enough —
but a fixed hand-rule is still worth building first as the thing to beat.

One-paragraph frame: For a content reviewer at a small team, deciding which pages to check
first, we will build a ranked priority queue from search and engagement signals, scoring pages
by likelihood of decline over a future window, measured by precision@K. A wrong call costs
either wasted review time or a missed real decline. A plain rule isn't enough because the
signals are individually noisy but meaningful combined. We will claim only observed,
directional, decision-support results — never a causal or algorithm-prediction claim.

Important flag for later weeks: the starter dataset's trend_direction label is rule-defined
(current-window bucket), not an observed future outcome — useful for this week's quick numbers,
but per the framing-ml-problems skill, my real target later should be a future-window label
(e.g. prior 90 days of features → decline over the next 30 days), not this proxy.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [4]:
!git clone https://github.com/larryjay007/MyML.git
%cd MyML

Cloning into 'MyML'...
remote: Enumerating objects: 102, done.
remote: Counting objects: 100% (102/102), done.
remote: Compressing objects: 100% (73/73), done.
remote: Total 102 (delta 23), reused 80 (delta 13), pack-reused 0 (from 0)
Receiving objects: 100% (102/102), 1.84 MiB | 12.71 MiB/s, done.
Resolving deltas: 100% (23/23), done.
/content/MyML/MyML


In [5]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

total_pages = len(df)
visible_pages = (df["impressions_90d"] > 0).sum()
declining_pages = (df["trend_direction"] == "down").sum()
declining_pct = declining_pages / total_pages

print(f"Total pages in starter dataset: {total_pages}")
print(f"Pages with measurable impressions (impressions_90d > 0): {visible_pages}")
print(f"Pages currently flagged as declining (trend_direction == 'down'): {declining_pages}")
print(f"Share of all pages that are declining: {declining_pct:.1%}")

Total pages in starter dataset: 30000
Pages with measurable impressions (impressions_90d > 0): 30000
Pages currently flagged as declining (trend_direction == 'down'): 16262
Share of all pages that are declining: 54.2%


Of the 30,000 pages in the starter dataset, 100% have measurable impressions in the last 90
days — this dataset is pre-filtered to visible pages only, so there's no "zero-traffic" noise
to filter out. More importantly: 16,262 pages (54.2%) are currently flagged as declining.
That's over half the inventory — far more than any small team could manually review one by one.
This is exactly the capacity problem Lane 2 addresses: not "is there a problem," but "which of
these 16,000+ pages should a reviewer look at first."

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

What this work can say: this is an observed pattern in a 30,000-page anonymized sample — over
half the pages show a current-window decline flag. Any ranking I build will be directional and
decision-support only: it prioritizes which pages a human reviewer should check first, based on
signals that historically correlate with decline or opportunity. It does not decide anything on
its own.

What this work can never say: it cannot claim to have proven why any page is declining, cannot
claim a refresh will cause recovery (that would require a real experiment, not this data), and
cannot claim anything about Google's actual ranking algorithm — search visibility signals are
observed correlations, not insight into how a search engine works internally. It also cannot
treat the current trend_direction label as ground truth forever — per the framing-ml-problems
skill, that label is rule-defined from the current window, not an observed future outcome, so
any claim built on it stays a proxy-based observation, not a validated prediction, until a
future-window label replaces it in later weeks.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.